interfaces/IDeviceInfo.h

```diff
@@ -25,136 +25,315 @@
+    };
+
+    /* @json 1.0.0 */
+    struct IImageMetadata : virtual Core::IUnknown {
+        enum { ID = ID_IMAGE_METADATA };
+
+        virtual ~IImageMetadata() override = default;
+
+        struct Firmware {
+            enum Yocto : uint8_t {
+                dunfell,
+                morty,
+                daisy,
+                kirkstone
+            };
+
+            string imageName /* @brief Name of firmware image */;
+            string sdk /* @brief SDK version string */;
+            string mediarite /* @brief Mediarite value */;
+            Yocto yocto /* @brief Yocto version */;
+        };
+
+        // @property
+        // @brief Retrieves Firware Information
+        virtual Core::hresult FirmwareInfo(Firmware& firmware /* @out */) const = 0;
+    };
+
+    /* @json 1.0.0 */
+    struct ISystemMetadata : virtual Core::IUnknown {
+        enum { ID = ID_SYSTEM_METADATA };
+
+        virtual ~ISystemMetadata() override = default;
+
+        struct System {
+            struct CPULoadAvgs {
+                uint64_t avg1min /* @brief 1min cpuload average */;
+                uint64_t avg5min /* @brief 5min cpuload average */;
+                uint64_t avg15min /* @brief 15min cpuload average */;
+            };
+            string version /* @brief Software version (in form *version#hashtag*) */;
+            uint64_t uptime /* @brief System uptime (in seconds) */;
+            uint64_t totalRAM /* @brief Total installed system RAM memory (in bytes) */;
+            uint64_t freeRAM /* @brief Free system RAM memory (in bytes) */;
+            uint64_t totalSwap /* @brief Total swap space (in bytes) */;
+            uint64_t freeSwap /* @brief Swap space still available (in bytes) */;
+            string deviceName /* @brief Host name */;
+            string cpuLoad /* @brief Current CPU load (percentage) */;
+            CPULoadAvgs cpuLoadAvg /* @brief CPU load average */;
+            string serialNumber /* @brief Device serial number */;
+            string time /* @brief Current system date and time */;
+        };
+
+        // @property
+        // @brief Retrieves SystemInfo
+        virtual Core::hresult SystemInfo(System& system /* @out */) const = 0;
+    };
+
+    /* @json 1.0.0 */
+    struct ISocketMetadata : virtual Core::IUnknown {
+        enum { ID = ID_SOCKET_METADATA };
+
+        virtual ~ISocketMetadata() override = default;
+
+        struct Socket {
+            uint32_t total /* @brief Number of total */;
+            uint32_t open /* @brief Number of open */;
+            uint32_t link /* @brief Number of link */;
+            uint32_t exception /* @brief Number of exception */;
+            uint32_t shutdown /* @brief Number of shutdown */;
+            uint32_t runs /* @brief Number of runs */;
+        };
+
+        // @property
+        // @brief Retrieves SocketInfo
+        virtual Core::hresult SocketInfo(Socket& socket /* @out */) const = 0;
```

Comment:
Maybe for consistency, if possible, call these methods like the interface bu than without the I, so:
virtual Core::hresult SocketMetadata(Socket& device /* @out */) const = 0;

Reply:
done